cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java

```diff
@@ -203,9 +206,24 @@ private Workflow compileWorkFlow(List<Map<String, Map<String, Object>>> workFlow
             String message = "Task: " + taskName + " syntax is illegal.\nBelow task name, there should be a map of values in the format:\ndo:\n\top_name:";
             try {
                 taskRawDataValue = taskRawData.values().iterator().next();
-                if (MapUtils.isNotEmpty(taskRawDataValue) && taskRawDataValue.containsKey(ScoreLangConstants.LOOP_KEY)) {
-                    message = "Task: " + taskName + " syntax is illegal.\nBelow the 'loop' keyword, there should be a map of values in the format:\nfor:\ndo:\n\top_name:";
-                    taskRawDataValue.putAll((Map<String, Object>) taskRawDataValue.remove(ScoreLangConstants.LOOP_KEY));
+                if (MapUtils.isNotEmpty(taskRawDataValue)) {
+                    boolean loopKeyFound = taskRawDataValue.containsKey(LOOP_KEY);
+                    boolean asyncLoopKeyFound = taskRawDataValue.containsKey(ASYNC_LOOP_KEY);
+                    if (loopKeyFound) {
+                        if (asyncLoopKeyFound) {
+                            throw new RuntimeException("Task: " + taskName + " syntax is illegal.\nBelow task name, there can be either \'loop\' or \'aync_loop\' key.");
+                        }
+
+                        message = "Task: " + taskName + " syntax is illegal.\nBelow the 'loop' keyword, there should be a map of values in the format:\nfor:\ndo:\n\top_name:";
+                        @SuppressWarnings("unchecked") Map<String, Object> loopRawData = (Map<String, Object>) taskRawDataValue.remove(LOOP_KEY);
+                        taskRawDataValue.putAll(loopRawData);
+                    }
+                    if (asyncLoopKeyFound) {
+                        message = "Task: " + taskName + " syntax is illegal.\nBelow the 'async_loop' keyword, there should be a map of values in the format:\nfor:\ndo:\n\top_name:";
+                        @SuppressWarnings("unchecked") Map<String, Object> asyncLoopRawData = (Map<String, Object>) taskRawDataValue.remove(ASYNC_LOOP_KEY);
```

Comment:
aren't you just putting it in the same exact place it was before?
it was on the ASYNC_LOOP_KEY under the root level and you put it back there again?